File: jak-daleko-vozilla-webapp/src/data/Location.ts
import { IObjectLocation } from "./IObjectsResponse";

export class Location {
    public latitude: number;
    public longitude: number;

    constructor(objectLocation: IObjectLocation) {
        this.latitude = objectLocation.latitude;
        this.longitude = objectLocation.longitude;
    }

    public isInBoundingBox(topLeft: Location, bottomRight: Location): boolean {
        return this.latitude >= topLeft.latitude &&
            this.longitude >= topLeft.longitude &&
            this.latitude <= bottomRight.latitude &&
            this.longitude <= bottomRight.longitude;
    }
}

File: jak-daleko-vozilla-webapp/Model/Location.cs
namespace jak_daleko_vozilla_webapp.Model
{
    public class Location
    {
        public Location(double latitude, double longitude)
        {
            Latitude = latitude;
            Longitude = longitude;
        }

        public double Latitude { get;  }
        public double Longitude { get;  }     
        
        public bool IsInBoundingBox(Location topLeft, Location bottomRight)
        {
            return Latitude <= topLeft.Latitude &&
                Longitude >= topLeft.Longitude &&
                Latitude >= bottomRight.Latitude &&
                Longitude <= bottomRight.Longitude;
        }
    }
}
